CMakeLists.txt

```diff
@@ -81,11 +81,26 @@ endif(Python_EXECUTABLE AND NOT PYTHON_INSTDIR)
 ########################################################################
 # Checks what linear algebra packages are installed                    #
 ########################################################################
-set(MKL_THREAD_LAYER "Intel OpenMP" CACHE STRING "The thread layer to choose for MKL")
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
+  set(MKL_THREAD_LAYER "Intel OpenMP" CACHE STRING "The thread layer to choose for MKL")
+else()
+  set(MKL_THREAD_LAYER "GNU OpenMP" CACHE STRING "The thread layer to choose for MKL")
+endif()
 find_package(MKL)
-if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
-  set_package_properties(MKL PROPERTIES TYPE OPTIONAL PURPOSE "Enables accelerated performance with MKL")
-endif (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+set_package_properties(MKL PROPERTIES TYPE OPTIONAL PURPOSE "Enables accelerated performance with MKL")
+
+#openmp with clang smaller version 10 and user defined reductions is buggy https://bugs.llvm.org/show_bug.cgi?id=44134
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 10.0)
+  # in case OpenMP was detected somewhere else
+  unset(OpenMP_FOUND)
+elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Intel" AND MKL_FOUND)
+  # part of mkl
+  unset(OpenMP_FOUND)
```

Comment:
I do not get this part. So if MKL_FOUND then OpenMP is also found but we have to check its version is larger than 4.5, but MKL_FOUND can be true with gcc as well. Why do we unset(OpenMP_FOUND)?

Reply:
In the case of `gcc` or `clang`, `find_package(MKL)` will call `find_package(OpenMP)` as well, but not for version 4.5 so we have to detect it again.

For intel when using MKL we don't want to detect OpenMP, hence we unset `OpenMP_FOUND` just in case `find_package(OpenMP)` was called somewhere else (same for clang <=10) as we are using `OpenMP_FOUND` later to link `OpenMP::OpenMP_CXX`.

Reply:
but do we not have  link against OpenMP::OpenMP_CXX even when using icc?

Reply:
Didn't you say there was a conflict between the two or was that only true for gcc?

Anyhow, I dropped that case, let's we what breaks.